plugins/Ntuplizer.cc

```diff
@@ -522,211 +553,247 @@ Ntuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
     //// W kinematic reconstruction
-    //  float pz = Utilities::RecoverNeutrinoPz(&TightMuonVect.at(0).p4(), &MET.p4());
-    //  Neutrino.setP4(reco::Particle::LorentzVector(MET.px(), MET.py(), pz, sqrt(MET.pt()*MET.pt() + pz*pz) ));
-    //  theW.addDaughter(TightMuonVect.at(0));
-    //  theW.addDaughter(Neutrino);
-    //  addP4.set(theW);
-
-    //      // SF
-    //if(isMC) {
-    //	//float LeptonWeightUnc = 0.;
-    //	LeptonWeight    *= theMuonAnalyzer->GetMuonTriggerSFIsoMu24(TightMuonVect.at(0));
-    //	//LeptonWeight    *= theMuonAnalyzer->GetMuonTrkSF(TightMuonVect.at(0));
-    //	LeptonWeight    *= theMuonAnalyzer->GetMuonIdSF(TightMuonVect.at(0), 0);
-    //	LeptonWeight    *= theMuonAnalyzer->GetMuonIsoSF(TightMuonVect.at(0), 0);
-    //
-    ////             LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonTriggerSFErrorMu50(TightMuonVect.at(0)),2);
-    ////             //LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonTrkSFError(TightMuonVect.at(0))        ,2);
-    ////             LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonIdSFError(TightMuonVect.at(0), 0)      ,2);
-    ////             LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonIsoSFError(TightMuonVect.at(0), 0)     ,2);
-    ////
-    ////             LeptonWeightUp   = LeptonWeight+sqrt(LeptonWeightUnc);
-    ////             LeptonWeightDown = LeptonWeight-sqrt(LeptonWeightUnc);
-    //}
-    //}
-    //    else if(isWtoEN) {
-
-    //      // W kinematic reconstruction
-    //float pz = Utilities::RecoverNeutrinoPz(&TightElecVect.at(0).p4(), &MET.p4());
-    //Neutrino.setP4(reco::Particle::LorentzVector(MET.px(), MET.py(), pz, sqrt(MET.pt()*MET.pt() + pz*pz) ));
-    //theW.addDaughter(TightElecVect.at(0));
-    //theW.addDaughter(Neutrino);
-    // addP4.set(theW);
-
-    //if(isMC) {
-    //	LeptonWeight    *= theElectronAnalyzer->GetElectronTriggerSFEle27Tight(TightElecVect.at(0));
-    //	LeptonWeight    *= theElectronAnalyzer->GetElectronRecoEffSF(TightElecVect.at(0));
-    //	LeptonWeight    *= theElectronAnalyzer->GetElectronIdSF(TightElecVect.at(0), 0);
-    //}
-    //}
-    //else if(isTtoEM) {
+      //     float pz = Utilities::RecoverNeutrinoPz(&TightMuonVect.at(0).p4(), &MET.p4());
+     //     Neutrino.setP4(reco::Particle::LorentzVector(MET.px(), MET.py(), pz, sqrt(MET.pt()*MET.pt() + pz*pz) ));
+     theW.addDaughter(TightMuonVect.at(0));
+     theW.addDaughter(MET);
+     addP4.set(theW);
+
+     if (theW.mt()<100. && isControl){
+       return;
+     }
+         // SF
+     if(isControl && isMC && !is2016) {
+      float LeptonWeightUnc = 0.;
+      LeptonWeightUp = 0.;
+      LeptonWeightDown = 0.;
+      //      LeptonWeight    *= theMuonAnalyzer->GetMuonTriggerSFIsoMu24(TightMuonVect.at(0));
+      //LeptonWeight    *= theMuonAnalyzer->GetMuonTrkSF(TightMuonVect.at(0));
+      LeptonWeight    *= theMuonAnalyzer->GetMuonIdSF(TightMuonVect.at(0), 1);
+      LeptonWeight    *= theMuonAnalyzer->GetMuonIsoSF(TightMuonVect.at(0), 1);
```

Comment:
Same comment as above: Vector filled with tight muons (i.e. `id=3`) but SF computed for `id=1`.